LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -833,7 +833,7 @@ internal async Task<bool> UpdateAfterJoinAsync(LoRaDeviceJoinUpdateProperties up
                     reportedProperties[TwinProperty.RX1DROffset] = null;
                 }
 
-                if (this.DesiredRX2DataRate != DefaultJoinValues && currentRegion.RegionLimits.IsCurrentDRIndexWithinAcceptableValue(this.DesiredRX2DataRate))
+                if (this.DesiredRX2DataRate != DefaultJoinValues && currentRegion.DRtoConfiguration.ContainsKey(this.DesiredRX2DataRate))
```

Comment:
couldn't we keep that validation in the RegionLimits method? I liked the encapsulation we had before without having to tangle with the dictionary here directly. If it is not really fitting the RegionLimits anymore, we may be able to add a method to the Region directly encapsulating that lookup?

Reply:
Looks like you already did that: IsCurrentDRIndexWithinAcceptableValue. Can we use that instead here and the other places where the direct lookups take place?

Reply:
The problem is that the input there is the DR index (uint) and my dictionary is a string. I fixed this in my previous implementation by looking at the dictionary size, which was sadly one of the problem causing this bug. I could implement it within the limit class, but then I would need to know the current region... Which defeat the idea of having a limit class imho. But happy to discuss on a call if you have other ideas.

Reply:
Wait I have an idea